packages/react-router/lib/server-runtime/routeModules.ts

```diff
@@ -1,95 +1,15 @@
-import type { Location } from "../router/history";
+import type { ActionFunction, LoaderFunction } from "../router/utils";
 import type {
-  ActionFunction as RRActionFunction,
-  ActionFunctionArgs as RRActionFunctionArgs,
-  AgnosticRouteMatch,
-  LoaderFunction as RRLoaderFunction,
-  LoaderFunctionArgs as RRLoaderFunctionArgs,
-  Params,
-} from "../router/utils";
-import type { AppData, AppLoadContext } from "./data";
-import type { SerializeFrom } from "../dom/ssr/components";
-import type { LinkDescriptor } from "../router/links";
+  ClientActionFunction,
+  ClientLoaderFunction,
+  LinksFunction,
+  MetaFunction,
+} from "../dom/ssr/routeModules";
 
 export interface RouteModules<RouteModule> {
   [routeId: string]: RouteModule | undefined;
 }
 
-/**
- * @deprecated Use `LoaderFunctionArgs`/`ActionFunctionArgs` instead
- */
-export type DataFunctionArgs = RRActionFunctionArgs<AppLoadContext> &
```

Comment:
All these can go away and we just use the RR types - and users will add `context: appLoadContext` via module augmentation: https://reactrouter.com/dev/guides/upgrading/remix#step-7---update-types-for-apploadcontext